eva_submission/eload_ingestion.py

```diff
@@ -114,3 +152,215 @@ def load_from_ena(self):
+            prop_files.append(str(properties_filename))
+        return prop_files
+
+    def get_mongo_creds(self):
+        properties = get_properties_from_xml_file('production', self.settings_xml_file)
+        mongo_host = str(str(properties['eva.mongo.host']).split(',')[0]).split(':')[0]
+        mongo_user = properties['eva.mongo.user']
+        mongo_pass = properties['eva.mongo.passwd']
+        return mongo_host, mongo_user, mongo_pass
+
+    def get_pg_creds(self):
+        properties = get_properties_from_xml_file('development', self.settings_xml_file)
+        pg_url = properties['eva.evapro.jdbc.url']
+        pg_user = properties['eva.evapro.user']
+        pg_pass = properties['eva.evapro.password']
+        return pg_url, pg_user, pg_pass
+
+    def get_pg_conn(self):
+        pg_url, pg_user, pg_pass = self.get_pg_creds()
+        # need to cut the jdbc: from the front of the pg_url
+        return psycopg2.connect(pg_url[5:], user=pg_user, password=pg_pass)
+
+    def get_study_name(self):
+        with self.get_pg_conn() as conn:
+            query = f"SELECT title FROM evapro.project WHERE project_accession='{self.project_accession}';"
+            rows = get_all_results_for_query(conn, query)
+        if len(rows) != 1:
+            raise ValueError(f'More than one project with accession {self.project_accession} found in metadata DB.')
+        return rows[0][0]
+
+    def get_vep_species(self):
+        words = self.eload_cfg.query('submission', 'scientific_name').lower().split()
+        return '_'.join(words)
+
+    def run_accession_workflow(self):
+        output_dir = self.create_nextflow_temp_output_directory(base=self.project_dir)
+        accession_config = {
+            'project_accession': self.project_accession,
+            'accession_props': self.eload_cfg.query(self.config_section, 'accession', 'properties'),
+            'public_dir': os.path.join(self.project_dir, project_dirs['public']),
+            'logs_dir': os.path.join(self.project_dir, project_dirs['logs']),
+            'executable': cfg['executable'],
+            'jar': cfg['jar'],
+        }
+        accession_config_file = os.path.join(self.project_dir, 'accession_config_file.yaml')
+        with open(accession_config_file, 'w') as open_file:
+            yaml.safe_dump(accession_config, open_file)
+        accession_script = os.path.join(ROOT_DIR, 'nextflow', 'ingestion.nf')
+        try:
+            command_utils.run_command_with_output(
+                'Nextflow Accessioning process',
+                ' '.join((
+                    'export NXF_OPTS="-Xms1g -Xmx8g"; ',
+                    cfg['executable']['nextflow'], accession_script,
+                    '-params-file', accession_config_file,
+                    '-work-dir', output_dir
+                ))
+            )
+        except subprocess.CalledProcessError as e:
+            self.error('Nextflow accessioning pipeline failed: results might not be complete')
```

Comment:
Could  the Nextflow log file be added to the error message for debugging purposes?